File: src/test/java/ru/uris/test/TestClient.java
package ru.uris.test;

import ru.uris.Client;
import ru.uris.Packet;

import java.io.IOException;
import java.util.Arrays;

public class TestClient {
    public static void main(String[] args) throws IOException {
        try (var client = new Client("localhost", 2014)) {
            client.objectPool.add(new Object());

            client.createListener().start();

            System.out.println(Arrays.toString(((Packet.PObjectList) client.sendAndReceive(new Packet.PObjectList(Packet.nextId()))).objects));
            System.out.println(Arrays.toString(((Packet.PMethodList) client.sendAndReceive(new Packet.PMethodList(Packet.nextId(), 0, true))).methods));
            System.out.println();
            for (var method : ((Packet.PMethodList) client.sendAndReceive(new Packet.PMethodList(Packet.nextId(), 0, true))).methods)
                System.out.println("[NAME]" + method.name + "\n[ARGS]:" + Arrays.toString(method.args) + "\n[RETURN]" + method.ret + "\n");
        }
    }
}

File: src/test/kotlin/ru/kuris/test/TestClient.kt
package ru.kuris.test

import kotlinx.coroutines.runBlocking
import ru.kuris.Client
import ru.kuris.Packet
import ru.kuris.Packet.Companion.nextId

object TestClient {
    @JvmStatic
    fun main(args: Array<String>) = runBlocking {
        Client("localhost", 2014).use { client ->
            client.objectPool.add(Any())
            client.createListener().start()
            println((client.sendAndReceive(Packet.Companion.PObjectList(nextId())) as Packet.Companion.PObjectList).objects.contentToString())
            println(
                (client.sendAndReceive(
                    Packet.Companion.PMethodList(
                        nextId(),
                        0,
                        true
                    )
                ) as Packet.Companion.PMethodList).methods.contentToString()
            )
            println()
            for (method in (client.sendAndReceive(
                Packet.Companion.PMethodList(
                    nextId(),
                    1,
                    true
                )
            ) as Packet.Companion.PMethodList).methods)
                println(
                    """
                    [NAME]${method.name}
                    [ARGS]:${method.args.contentToString()}
                    [RETURN]:${method.ret}

                    """.trimIndent()
                )
        }
    }
}